packages/rich-text/src/plugins/Tracking/createTrackingPlugin.ts

```diff
@@ -111,5 +111,55 @@ export const createTrackingPlugin = (onAction: RichTextTrackingActionHandler): R
 
       return editor;
     },
+    then: (editor) => ({
+      editor: {
+        insertData: {
+          format: 'text/html',
+          getFragment: ({ data }) => {
+            let source;
+
+            if (data.includes('docs-internal-guid')) {
+              source = 'Google Docs';
+            }
+
+            if (data.includes('<google-sheets-html-origin>') || data.includes('data-sheets-')) {
+              source = 'Google Spreadsheets';
+            }
+
+            // TODO: MS Word Online doesn't give us specific tags, we might need to have a closer look at its tracking result since we are using generic values to identify it
+            if (
+              data.includes('Arial_MSFontService') &&
+              (data.includes('class="TextRun') || data.includes('class="OutlineElement'))
+            ) {
+              source = 'Microsoft Word Online';
+            }
+
+            if (data.includes('<meta name=Generator content="Microsoft Excel')) {
```

Comment:
A more reliable way I think is to use `DomParser().parseFromString` then you could use [CSS selectors like what we do in the HTML plugin](https://github.com/contentful/field-editors/blob/84825818e6684c80cf9862f62d197f77ef54a8f8/packages/rich-text/src/plugins/PasteHTML/utils/sanitizeSheets.ts#L11-L12). Same goes for other `.includes()` calls.